scripts/irods/test/test_iadmin.py

```diff
@@ -2539,23 +2539,32 @@ def test_mkgroup_with_remote_zone(self):
 
 class test_zone_conn_str_validation(unittest.TestCase):
 
-    def test_mkzone_bad(self):
-        self.admin.assert_icommand_fail(['iadmin', 'mkzone', 'badzone1', 'remote', 'hostname'],
-                                        'STDERR', 'CAT_INVALID_ARGUMENT')
-        self.admin.assert_icommand_fail(['iadmin', 'mkzone', 'badzone2', 'remote', 'hostname:0'],
-                                        'STDERR', 'CAT_INVALID_ARGUMENT')
-        self.admin.assert_icommand_fail(['iadmin', 'mkzone', 'badzone3', 'remote', ':1234'],
-                                        'STDERR', 'CAT_HOSTNAME_INVALID')
-        #self.admin.assert_icommand_fail(['iadmin', 'mkzone', 'badzone4', 'remote', ''],
-        #                                'STDERR', 'CAT_INVALID_ARGUMENT')
-        self.admin.assert_icommand_fail(['iadmin', 'mkzone', 'badzone5', 'remote', 'hostname:1234567890'],
-                                        'STDERR', 'CAT_INVALID_ARGUMENT')
-        self.admin.assert_icommand_fail(['iadmin', 'mkzone', 'badzone6', 'remote', 'hostname:asdf1234asdf'],
-                                        'STDERR', 'CAT_INVALID_ARGUMENT')
-        self.admin.assert_icommand_fail(['iadmin', 'mkzone', 'badzone7', 'remote', 'hostname:asdf1234'],
-                                        'STDERR', 'CAT_INVALID_ARGUMENT')
-        self.admin.assert_icommand_fail(['iadmin', 'mkzone', 'badzone8', 'remote', 'hostname:1234asdf'],
-                                        'STDERR', 'CAT_INVALID_ARGUMENT')
-        self.admin.assert_icommand_fail(['iadmin', 'mkzone', 'badzone9', 'remote', 'hostname:-1234'],
-                                        'STDERR', 'CAT_INVALID_ARGUMENT')
+    @classmethod
+    def setUpClass(cls):
+        """Set up the test class."""
+        cls.admin = session.mkuser_and_return_session('rodsadmin', 'otherrods', 'rods', lib.get_hostname())
```

Comment:
Is there a reason you chose to do this instead of something like the following:
https://github.com/irods/irods/blob/25d8f67df6c123ee6f0b3fe4695e4042b52cdb59/scripts/irods/test/test_ichmod.py#L14

Reply:
Yes. The mixin uses instance initializers instead of class initializers, which isn't needed for this test case.